### DISCUSSION TITLE
Simultaneous "Selected only" & "Non-Selected only" in Object Selection

### DISCUSSION THREAD

**Discussion by Richard-L** [OpenRCT2/OpenRCT2#15736] [Feature requests]
OpenRCT2, v0.2.5-127-g0cbd70029 (0cbd700 on develop) provided by GitHub
Windows 7 (x86-64)

![image](https://user-images.githubusercontent.com/4934209/79132624-f5349e00-7daa-11ea-9324-f7f39dd96242.png)

While I have a hunch there's a good reason as to why this isn't already possible, …would you consider allowing both of these options checked simultaneously? It's useful to see what the current `object` folder holds and lets you choose against something you already have in your bench.

It's especially troublesome when using "Selected only" and accidentally click on something that then immediately disappears from the list. You then have to change to "Non-selected only" and find the respective item again.

**Comment by jensj12** [comment]
It's in the word 'only'. You can deselect the currently selected option to view both selected and non-selected objects.

**Comment by Richard-L** [comment]
I see, thanks for explaining.

The logic of the menu could be improved then. You are offering a checkbox button where a radio button would need to be.

I suggest you take out "Selected only" and "Non-selected only" from the filter dropdown, and instead make it a radio button choice to the left or right of filter, with three choices (1) "Selected", (2) "Non-selected" and (3) "Both".

Looking now, I'm afraid there are no radio buttons in Open, ever, so in that case the choice might be a dropdown with the three options.

Not directly related to this particular issue, but can you explain what the "advanced" button is for? I've never found myself _not_ wanting to see all panels required. So the first thing I do whenever I use object selection is to click "advanced", anyway.

**Comment by Umdlye** [comment]
The advanced button is probably there to avoid overwhelming casual players with too many choices.  They would just select a bunch of scenery groups for their scenario and move on.
